features/menu/lib/bloc/menu_bloc.dart

```diff
@@ -0,0 +1,98 @@
+import 'package:domain/domain.dart';
+import 'package:core/core.dart';
+
+part 'menu_state.dart';
+part 'menu_event.dart';
+
+// class CombinedMenuItems {
+//   final List<MenuBloc> menuBloc;
+//   final List<HorizontalMenuBloc> horizontalMenuBloc;
+//
+//   CombinedMenuItems(this.menuBloc, this.horizontalMenuBloc);
+// }
+
+
+class HeaderMenuBloc extends Bloc<HeaderMenuEvent, HeaderMenuState> {
+  final GetHeaderMenuListUseCase _getHeaderMenuListUseCase;
+
+  HeaderMenuBloc({
+    required GetHeaderMenuListUseCase getHeaderMenuListUseCase,
+  }) : _getHeaderMenuListUseCase = getHeaderMenuListUseCase,
+        super(const HeaderMenuState()) {
+    on<LoadHeaderMenuList>(_getHeaderMenu);
+    add(LoadHeaderMenuList(0));
+  }
+
+  Future<void> _getHeaderMenu(LoadHeaderMenuList event, Emitter<HeaderMenuState> emit) async {
+    try {
+      emit(
+        state.copyWith(isLoading: true, error: null),
+      );
+      final List<HeaderMenuItemEntity> dishes =
+      await _getHeaderMenuListUseCase.execute(event.page);
+      emit(state.copyWith(
+          imagesList: [...state.imagesList, ...dishes], isLoading: false));
+    } catch (e) {
+      emit(
+        state.copyWith(error: e, isLoading: false),
+      );
+    }
+  }
+}
+
+class HorizontalMenuBloc extends Bloc<HorizontalMenuEvent, HorizontalMenuState> {
+  final GetHorizontalMenuListUseCase _getHorizontalMenuListUseCase;
+
+  HorizontalMenuBloc({
+    required GetHorizontalMenuListUseCase getHorizontalMenuListUseCase,
+  }) : _getHorizontalMenuListUseCase = getHorizontalMenuListUseCase,
+       super(const HorizontalMenuState()) {
+    on<LoadHorizontalMenuList>(_getHorizontalMenu);
+    add(LoadHorizontalMenuList(0));
+  }
+
+  Future<void> _getHorizontalMenu(LoadHorizontalMenuList event, Emitter<HorizontalMenuState> emit) async {
+    try {
+      emit(
+        state.copyWith(isLoading: true, error: null),
+      );
+      final List<HorizontalMenuItemEntity> dishes =
+      await _getHorizontalMenuListUseCase.execute(event.page);
+      emit(state.copyWith(
+          dishesList: [...state.dishesList, ...dishes], isLoading: false));
+    } catch (e) {
+      emit(
+        state.copyWith(error: e, isLoading: false),
+      );
+    }
+  }
+}
+
+
+class MenuBloc extends Bloc<MenuEvent, MenuState> {
```

Comment:
Why we use 3 blocs in one file? What is it?

Reply:
In this file we can see 3 block, where HeaderMenuBloc is responsible for managing the state of images on the Single Restaurant Screen, HorizontalMenuBloc for managing images on the horizontal food  list, bottom on the restaurant properties. And MenuBloc for managing vertical food list. I think that I did the wrong naming in the classes, or the solution can be simplified.